cmd/kube-controller-manager/app/certificates.go

```diff
@@ -39,10 +39,14 @@ import (
 )
 
 func startCSRSigningController(ctx ControllerContext) (http.Handler, bool, error) {
-	if !ctx.AvailableResources[schema.GroupVersionResource{Group: "certificates.k8s.io", Version: "v1beta1", Resource: "certificatesigningrequests"}] {
+	gvr := schema.GroupVersionResource{Group: "certificates.k8s.io", Version: "v1beta1", Resource: "certificatesigningrequests"}
+	if !ctx.AvailableResources[gvr] {
+		klog.Warningf("Resource %s is not available now", gvr.String())
 		return nil, false, nil
 	}
 	if ctx.ComponentConfig.CSRSigningController.ClusterSigningCertFile == "" || ctx.ComponentConfig.CSRSigningController.ClusterSigningKeyFile == "" {
+		// normally it should not go here since we have applied default values
+		klog.Warningf("Empty value. Please check --cluster-signing-cert-file and --cluster-signing-key-file")
```

Comment:
remove this warning. we should *not* be setting a default for the signer CA/key. we'll have to fix that in v1beta1 of the controller config, but in the meantime, don't presume that these fields are defaulted.

Reply:
see the warning on line 76 about defaulting